Assets/MixedRealityToolkit.Tools/Experimental/MigrationWindow/MigrationWindow.cs

```diff
@@ -230,7 +231,9 @@ private void SetMigrationHandlerType()
         {
             try
             {
-                selectedMigrationHandlerType = Type.GetType(migrationHandlerTypeNames[selectedMigrationHandlerIndex], true);
+                selectedMigrationHandlerType = AppDomain.CurrentDomain.GetAssemblies()
+                                               .SelectMany(x => x.GetTypes())
```

Comment:
I think that we'll want to use GetLoadableTypes() here instead of GetTypes() to avoid cases where some binaries have types that aren't loadable:

https://github.com/microsoft/MixedRealityToolkit-Unity/blob/f38399007d30bcee7d3e7ceb27c058074650a311/Assets/MixedRealityToolkit/Extensions/AssemblyExtensions.cs#L16

Reply:
Changing this. I made the choice to throw and log on catch but I guess I can just log if null. Apart from getting the benefit of the Extension, would you say it would be bad practice to choose to log in the catch?

Reply:
Oh nah, it's a fine practice in general. The tricky part that the previous code could throw an exception because there was some random assembly that totally borked all of the type loading code - i.e. if ANY assembly/binary that was being enumerated over threw an exception, the entire type loading code would fail. So there could very well be a valid type to migrate (or a number of valid things to migrate), but the mere presence of a broken/weird assembly would mess up everything.

That's the main danger of the code before, which is why using GetLoadableTypes() is important.

Reply:
FWIW ideally if we had some other static analysis tools we could just error out whenever we saw this happening (otherwise it is pretty hard to catch except via manual CR).